tests/fork/SingleTokenConverter.ts

```diff
@@ -0,0 +1,171 @@
+import chai from "chai";
+import { Contract, Signer } from "ethers";
+import { parseUnits } from "ethers/lib/utils";
+import hre, { network } from "hardhat";
+
+const { expect } = chai;
+
+const initMainnetUser = async (user: string) => {
+  await impersonateAccount(user);
+  return ethers.getSigner(user);
+};
+
+export async function setForkBlock(blockNumber: number) {
+  await network.provider.request({
+    method: "hardhat_reset",
+    params: [
+      {
+        forking: {
+          jsonRpcUrl: process.env.BSC_ARCHIVE_NODE,
+          blockNumber: blockNumber,
+        },
+      },
+    ],
+  });
+}
+
+const forking = (blockNumber: number, fn: () => void) => {
+  describe(`riskFundConverter #${blockNumber}`, () => {
+    before(async () => {
+      await setForkBlock(blockNumber);
+    });
+    fn();
+  });
+};
+
+async function impersonateAccount(accountAddress) {
+  await hre.network.provider.request({
+    method: "hardhat_impersonateAccount",
+    params: [accountAddress],
+  });
+}
+
+const NORMAL_TIMELOCK = "0x939bD8d64c0A9583A7Dcea9933f7b21697ab6396";
+const PROTOCOL_SHARE_RESERVE = "0xCa01D5A9A248a830E9D93231e791B1afFed7c446";
+const SINGLE_TOKEN_CONVERTER_BEACON_PROXY = "0x4c9D57b05B245c40235D720A5f3A592f3DfF11ca";
+const RISK_FUND_CONVERTER_PROXY = "0xA5622D276CcbB8d9BBE3D1ffd1BB11a0032E53F0";
+const PROXY_ADMIN = "0x6beb6D2695B67FEb73ad4f172E8E2975497187e4";
+const BTCB_PRIME_CONVERTER = "0xE8CeAa79f082768f99266dFd208d665d2Dd18f53";
+const CORE_POOL = "0xfd36e2c2a6789db23113685031d7f16329158384";
+
+//Assets listed in core pool and need to release mfunds for them
+const USDT = "0x55d398326f99059fF775485246999027B3197955";
+const BTCB = "0x7130d2A12B9BCbFAe4f2634d864A1Ee1Ce3Ead9c";
+
+const USDT_HOLDER = "0xf89d7b9c864f589bbF53a82105107622B35EaA40";
+
+async function getToken(tokenAddress) {
+  const token = await hre.ethers.getContractAt("MockToken", tokenAddress);
+  return token;
+}
+
+forking(35936683, () => {
+  let protocolShareReserve: Contract;
+  let singleTokenConverterBeaconProxy: Contract;
+  let proxyAdmin = Contract;
+  let singleTokenConverterImplementation = Contract;
+  let timeLockSigner: Signer;
+
+  if (process.env.FORK_MAINNET === "true") {
```

Comment:
If we put this in a separate top level folder we can remove the nesting under an environment variable. 

Fork tests can be imported into an index file in the root of this directory in the order to be run and then run with the command. 

`yarn hardhat test fork-tests/index.ts`

This makes running for tests more explicit, less error prone, should make working with fork and unit tests easier, and makes them easier to read

Reply:
I think it will be much better to keep each test file separate as every file contains tests for different purposes and if someone wants to run the fork tests only for a single file then it will be handy.

After all we can use the below command to run all the fork tests at once
`npx hardhat test tests/hardhat/Fork/*`

Also executing Unit tests separately from fork tests is better because hardhat sometimes consider the unit tests also as fork tests if we execute them together.

Reply:
Yeah for sure we can keep it this why my suggestion also meets the requirements of keeping tests in a separate file.

Good to know about the glob. Then the main difference is moving the fork tests out of the tests directory. Then you don't need the environment variable to prevent the fork tests from running with unit tests.